app/validation/validator.py

```diff
@@ -581,14 +621,14 @@ def convert_to_datetime(value):
 
     @staticmethod
     def _get_all_answer_ids(json_to_validate):
-        answer_ids = []
+        answer_ids = {}
 
         for section in json_to_validate.get('sections'):
             for group in section.get('groups'):
                 for block in group.get('blocks'):
                     if block.get('questions'):
                         for question in block['questions']:
                             for answer in question.get('answers', []):
-                                answer_ids.append(answer['id'])
+                                answer_ids[answer['id']] = {'group_id': group['id']}
```

Comment:
This has drastically changed the behaviour of this function so we should probably rename it?

Reply:
I also can't see why this change was made?

Reply:
Not really, still returns answer IDs, just with some metadata attached.  We pass around the group IDs so in repeat rules we can check the answer is in the same group.

Reply:
it feels like a bit of a hack just to get the group id for an answer in 1 function?
I don't think that the function name describes what is returned.